lib/src/solar.dart

```diff
@@ -0,0 +1,71 @@
+// ignore_for_file: unnecessary_overrides
+
+import 'lunar.dart';
+
+/// An instant in time, similar to [DateTime] object, having functions
+/// that helps interacting with [Lunar] much more easier.
+class Solar implements Comparable<Solar> {
+  late final int _year;
+  int get year => _year;
+
+  late final int _month;
+  int get month => _month;
+
+  late final int _day;
+  int get day => _day;
+
+  late final int _hour;
+  int get hour => _hour;
+
+  late final int _minute;
+  int get minute => _minute;
+
+  late final int _second;
+  int get second => _second;
+
+  /// Create an instance of [Solar] from [dateTime]. If [dateTime] is null,
+  /// automatically constructs a [Solar] with current date and time
+  /// in the local time zone.
+  Solar([DateTime? dateTime])
+      : this._fromDate(dateTime ?? DateTime.now().toLocal());
+
+  Solar._fromDate(DateTime dateTime) {
+    _year = dateTime.year;
+    _month = dateTime.month;
+    _day = dateTime.day;
+    _hour = dateTime.hour;
+    _minute = dateTime.minute;
+    _second = dateTime.second;
```

Comment:
you can parse get timezone as int by `timeZoneOffset.inHours` instead of use default timezone (07:00)
ex:
```dart
_timezone = dateTime.timeZoneOffset.inHours
```

Reply:
Reference my reply [here](https://github.com/nguyen703/vnlunar/pull/3#discussion_r1378528134)